src/app/src/main/kotlin/org/imperial/mrc/hint/FileManager.kt

```diff
@@ -65,9 +67,27 @@ class LocalFileManager(
 
         val resourceUrl = getResourceUrl(data, originalFilename)
 
-        val inputStream = adr.getInputStream(data.url)
+        val response = adr.getInputStream(data.url)
 
-        return saveFile(inputStream, originalFilename, type, true, resourceUrl)
+        if (response.statusCode() != 200)
+        {
+            /**
+             * When a user is unauthenticated or lack required permission, the user gets redirected
+             * to auth0 login page. handleAdrException handles permission and unexpected ADR errors
+             */
+            if (response.uri().host.contains("eu.auth0.com"))
```

Comment:
Is this really the only way we can tell if we've had an auth issue? Hardcoding the auth0 host seems really grimy! Could we at least just check if it's a redirect? Do we get a 302 or something here? ADR can't return 4xx?